### PR TITLE
[Kotlin Migration] UsageAnalytics

### PR DESCRIPTION
**Pull request by Akshay0701**

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/analytics/UsageAnalytics.kt [hunk 1]
```diff
@@ -57,400 +50,411 @@ public class UsageAnalytics {
...
-        sOriginalUncaughtExceptionHandler = null;
+    @Synchronized
+    private fun unInstallDefaultExceptionHandler() {
+        Thread.setDefaultUncaughtExceptionHandler(sOriginalUncaughtExceptionHandler)
+        sOriginalUncaughtExceptionHandler = null
     }
 
-
     /**
+     * Determine whether we are disabled or not
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/analytics/UsageAnalytics.kt:136]: Split the documentation between the get and the set, it's been combined.

File: AnkiDroid/src/main/java/com/ichi2/anki/analytics/UsageAnalytics.kt [hunk 2]
```diff
@@ -57,400 +50,411 @@ public class UsageAnalytics {
...
-
     /**
      * Submit a screen for aggregation / analysis.
      * Intended for use to determine if / how features are being used
      *
      * @param object the result of Object.getClass().getSimpleName() will be used as the screen tag
      */
-    public static void sendAnalyticsScreenView(Object object) {
-        sendAnalyticsScreenView(object.getClass().getSimpleName());
+    fun sendAnalyticsScreenView(`object`: Objects) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/analytics/UsageAnalytics.kt:188]: rename the variable (or add a `cleanup`)

File: AnkiDroid/src/main/java/com/ichi2/anki/analytics/UsageAnalytics.kt [hunk 3]
```diff
@@ -57,400 +50,411 @@ public class UsageAnalytics {
...
-
     /**
      * Send a detailed arbitrary analytics event, with noun/verb pairs and extra data if needed
      *
      * @param category the category of event, make your own but use a constant so reporting is good
      * @param action   the action the user performed
      * @param value    A value for the event, Integer.MIN_VALUE signifies caller shouldn't send the value
      * @param label    A label for the event, may be null
+     *
+     * Send an arbitrary analytics event - these should be noun/verb pairs, e.g. "text to speech", "enabled"
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/analytics/UsageAnalytics.kt:215]: Documentation was duplicated due to overloads. Should be combined

File: AnkiDroid/src/main/java/com/ichi2/anki/analytics/UsageAnalytics.kt [hunk 4]
```diff
@@ -57,400 +50,411 @@ public class UsageAnalytics {
...
-        Throwable cause;
-        Throwable result = t;
-
-        while (null != (cause = result.getCause()) && (!result.equals(cause))) {
-            result = cause;
+    @JvmStatic
+    fun getCause(t: Throwable): Throwable {
+        var cause: Throwable
+        var result = t
+        while (null != result.cause.also { cause = it!! } && result != cause) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/analytics/UsageAnalytics.kt:250]: This looks to be incorrect. 

Could you KotlinCleanup("unit test required") if it was incorrect

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
I think there's one issue which this flagged. Rest are nits

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
```
> Task :AnkiDroid:compilePlayReleaseKotlin FAILED
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/analytics/AnalyticsDialogFragment.kt: (24, 24): None of the following functions can be called with the arguments supplied: 
public final fun sendAnalyticsScreenView(`object`: Objects): Unit defined in com.ichi2.anki.analytics.UsageAnalytics
public final fun sendAnalyticsScreenView(screenName: String?): Unit defined in com.ichi2.anki.analytics.UsageAnalytics
```

**Comment by mikehardy** [comment]
A lot of these PRs are being rejected based on automated CI tests. Please do not push commits until you know they pass `./gradlew clean jacocoTestReport :api:lintRelease :AnkiDroid:lintPlayRelease ktlintCheck` locally - the review cycles are really expensive, the compute cycles are cheap

**Pull request closed without merging**